Core/Main/itkTransformixFilter.h

```diff
@@ -220,6 +221,19 @@ class ITK_TEMPLATE_EXPORT TransformixFilter : public ImageSource<TMovingImage>
     return m_OutputMesh;
   }
 
+  /** Sets the transformation. If null, the transformation is entirely specified by the transform
+   * parameter object that is set by SetTransformParameterObject. Otherwise, the transformation is specified by this
+   * transform object, with additional information from the specified transform parameter object. */
```

Comment:
Is this implementation not the same as the itkSetConstObjectMacro ?

Reply:
@mstaring Welll... it's similar, looking at https://github.com/InsightSoftwareConsortium/ITK/blob/49150c8a701f629a122646f708a802f961d1ace7/Modules/Core/Common/include/itkMacro.h#L1185-L1195
```
#define itkSetConstObjectMacro(name, type)             \
  virtual void Set##name(const type * _arg)            \
  {                                                    \
    itkDebugMacro("setting " << #name " to " << _arg); \
    if (this->m_##name != _arg)                        \
    {                                                  \
      this->m_##name = _arg;                           \
      this->Modified();                                \
    }                                                  \
  }                                                    \
```

Three differences:
- TransformixFilter::SetTransform is non-virtual. I don't think it should be overridden, otherwise it just might not work anymore. My proposal to remove the virtual keywords from those ITK Get/Set macro's is postponed until ITK 6.0.0: https://github.com/InsightSoftwareConsortium/ITK/pull/2785
- The parameter of TransformixFilter::SetTransform is a SmartPointer, indicating that the transform object will be "shared" between the caller and TransformixFilter. I think that makes sense.
- TransformixFilter::SetTransform does not call `itkDebugMacro`. Should it do so?

Anyway, if you prefer `itkSetConstObjectMacro`, I can adjust the code, no problem.

Reply:
OK, now using the itkSetConstObjectMacro, although I'm not very happy about the current implementation of the macro.